### DISCUSSION TITLE
Can RollForward.tip ever be 'origin'?

### DISCUSSION THREAD

**Discussion by mkazlauskas** [CardanoSolutions/ogmios#302] [Q & A] [answered]
According to TypeScript [types](https://github.com/CardanoSolutions/ogmios/blob/master/clients/TypeScript/packages/schema/src/index.ts#L1044), `tip` in `RollForward` event is `TipOrOrigin`. Since event contains a `block`, node must have at least 1 block. Shouldn't it be just `Tip`? If not, what is the scenario when it would be `'origin'`?

**Comment by KtorZ** [comment, marked as answer]
It can't. The only reason it's specified as such is because the type is reused in some other places where origin is a possible case. This should have been modeled as two different types but since I am also relying on input-output's codebase for some parts of the testing (reusing arbitrary generators for property-based testing), I needed to play along and keep that. There are a few other "inconsistency" like this one in the JSON schema which are a bit annoying and stem mainly from the underlying ledger/consensus Haskell representations.
